openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,198 @@
+        """
+        Method that creates a user in the ICC database based int the  current user logged in data.
+        Also takes care of duplicated usernames.
+        """
+        log_details = {
+            "url": settings.ICC_CREATE_USER_API_FUNCTION,
+        }
+        icc_user = {}
+
+        try:
+            user, _ = get_user(email=data.get('user_email'))
+            _data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_CREATE_USER_API_FUNCTION,
+                "users[0][username]": user.username + self._get_random_string(
+                    self.USERNAME_SUFFIX_LENGTH) if duplicated_username else user.username,
+                "users[0][password]": self._get_random_string(settings.ICC_HASH_LENGTH),
+                "users[0][firstname]": user.first_name,
+                "users[0][lastname]": user.last_name,
+                "users[0][email]": user.email,
+                "users[0][auth]": "saml2",
+            }
+
+            response = requests.post(
+                url=settings.ICC_BASE_URL,
+                data=_data,
+            )
+            log_details['request_payload'] = _data
+
+        except Exception as error:
+            LOG.error('Failed to create ICC user. Reason: %s', str(error))
+            log_details['request_payload']['wstoken'] = 'icc-api-token'
+            log_details['response'] = {'error': 'Failed to create ICC user. Reason: ' + str(error)}
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
+
+            pass
+        else:
+            icc_user = self._get_icc_user_from_xml_response(response, 'create_user')
+
+        return icc_user
+
+    def _get_icc_user_from_xml_response(self, response, method_type):
+        """
+        Method that receives a xml response and convert it to json object and returns it.
+        """
+        icc_user = {}
+
+        try:
+            str_response = xmltodict.parse(response.content)
+            json_str_response = json.dumps(str_response)
+            json_obj_response = json.loads(json_str_response)
+
+            if method_type == 'create_user':
+                fields = json_obj_response['RESPONSE']['MULTIPLE']['SINGLE']['KEY']
+            else:
+                fields = json_obj_response['RESPONSE']['SINGLE']['KEY'][0]['MULTIPLE']['SINGLE']['KEY']
+
+            for field in fields:
+                if field['@name'] == 'id':
+                    icc_user['id'] = field['VALUE']
+                if field['@name'] == 'username':
+                    icc_user['username'] = field['VALUE']
+                    break
+
+        except Exception:
+            pass
+
+        finally:
+            return icc_user
+
+    def _get_random_string(self, length):
+        """
+        Method that generates and return a random string
+        """
+        # TODO: uncommend return for production
```

Comment:
Please uncomment this method. @ivanvgh

Reply:
This is only for production environment, the string returned is for testing purpose, for stage environment asked by Scott.

Reply:
So when are you going to uncomment this block? @ivanvgh

Reply:
Once Scott finish testing this feature on Stage it is going to be changed before pushed to production.

Reply:
If this is going to be deployed to the stage environment, I do think, it's better to use Django settings to control this behaviour and avoid making changes after this is be tested on any other environment than local. @ivanvgh

Reply:
This method is used for a random password and a username suffix, then I added a validation when it is called to generate a suffix it return the random string in the other hand it will return site configuration value for the random password as recommended.